python/pyspark/pandas/series.py

```diff
@@ -4485,6 +4489,173 @@ def replace(
+
+             pandas-on-Spark uses return type hint and does not try to infer the type.
+
+        Parameters
+        ----------
+        other : Series or scalar
+            The value(s) to be combined with the `Series`.
+        func : function
+            Function that takes two scalars as inputs and returns an element.
+            Note that type hint for return type is strongly recommended.
+        fill_value : scalar, optional
+            The value to assume when an index is missing from
+            one Series or the other. The default specifies to use the
+            appropriate NaN value for the underlying dtype of the Series.
+
+        Returns
+        -------
+        Series
+            The result of combining the Series with the other object.
+
+        See Also
+        --------
+        Series.combine_first : Combine Series values, choosing the calling
+            Series' values first.
+
+        Examples
+        --------
+        Consider 2 Datasets ``s1`` and ``s2`` containing
+        highest clocked speeds of different birds.
+
+        >>> from pyspark.pandas.config import set_option, reset_option
+        >>> set_option("compute.ops_on_diff_frames", True)
+        >>> s1 = ps.Series({'falcon': 330.0, 'eagle': 160.0})
+        >>> s1
+        falcon    330.0
+        eagle     160.0
+        dtype: float64
+        >>> s2 = ps.Series({'falcon': 345.0, 'eagle': 200.0, 'duck': 30.0})
+        >>> s2
+        falcon    345.0
+        eagle     200.0
+        duck       30.0
+        dtype: float64
+
+        Now, to combine the two datasets and view the highest speeds
+        of the birds across the two datasets
+
+        >>> s1.combine(s2, max)
+        duck        NaN
+        eagle     200.0
+        falcon    345.0
+        dtype: float64
+
+        In the previous example, the resulting value for duck is missing,
+        because the maximum of a NaN and a float is a NaN.
+        So, in the example, we set ``fill_value=0``,
+        so the maximum value returned will be the value from some dataset.
+
+        >>> s1.combine(s2, max, fill_value=0)
+        duck       30.0
+        eagle     200.0
+        falcon    345.0
+        dtype: float64
+        >>> reset_option("compute.ops_on_diff_frames")
+        """
+        if not isinstance(other, Series) and not pd.api.types.is_scalar(other):
+            raise TypeError("unsupported type: %s" % type(other))
+
+        if not callable(func):
+            raise TypeError("%s object is not callable" % type(func).__name__)
+
+        if pd.api.types.is_scalar(other):
+            tmp_other_col = verify_temp_column_name(self._internal.spark_frame, "__tmp_other_col__")
+            combined = self.to_frame()
+            combined[tmp_other_col] = other
+            combined = DataFrame(combined._internal.resolved_copy)
+        elif same_anchor(self, other):
+            combined = self._psdf[self._column_label, other._column_label]
+        elif fill_value is None:
```

Comment:
quick question: What if same_anchor with `fill_value`?

Reply:
`fill_value` is only used when an index is missing from one Series or the other. With two series from same dataframe, there are no missing indices.
https://pandas.pydata.org/docs/reference/api/pandas.Series.combine.html

Reply:
Thanks for clarify, so we'd better also change the below test to cover this specific case, see it work or not?

Reply:
I put `fill_value` tests in python/pyspark/pandas/tests/test_ops_on_diff_frames.py